azure_monitor/src/azure_monitor/trace.py

```diff
@@ -87,7 +87,7 @@ def span_to_envelope(self, span):  # noqa pylint: disable=too-many-branches
                     span.context.span_id
                 ),
                 duration=utils.ns_to_duration(span.end_time - span.start_time),
-                responseCode="0",
+                responseCode=str(StatusCanonicalCode.OK.value),
```

Comment:
I mean we have a real status of the span, can we use that one?

Reply:
StatusCode might not exist in span attributes.

Reply:
I was trying to keep the old behaviour, but since there were a bunch of TODOs it's safe to assume that we want to change this. I will change it to use the status of the span.